### ISSUE TITLE
How the output is fed back to the decoder? 

### ISSUE THREAD

**Issue by harsh-agarwal** [tobyyouup/conv_seq2seq#5]
Hi,

I am a beginner in tensorflow so the doubt might be silly.

I was trying to work on a CAFFE variant of this architecture. As I assume there aren't really any complicated operations that seem to happen according to the paper neither according to your code. However, I am struggling to understand how would the output be fed back to the decoder. 

As far as I understood the output token that we find is concatenated to the input of the decoder till you reach the end of the sentence. But how has that been implemented? The code happens to have a step() function that does this concatenation, but that isn't called anywhere.

[This line of your code](https://github.com/tobyyouup/conv_seq2seq/blob/78a6e4e62a4c57a5caa9d584033a85e810fd726e/seq2seq/decoders/conv_decoder_fairseq.py#L197):

`def step(self, time, inputs, state, name=None):
   
    cur_inputs = inputs[:,0:time+1,:] 
    zeros_padding = inputs[:,time+2:,:] 
    cur_inputs_pos = self.add_position_embedding(cur_inputs, time)
    
    enc_output = state 
    logits = self.infer_conv_block(enc_output, cur_inputs_pos)
    
    sample_ids = tf.cast(tf.argmax(logits, axis=-1), dtypes.int32)

    finished, next_inputs = self.next_inputs(sample_ids=sample_ids)
    next_inputs = tf.reshape(next_inputs, [self.config.beam_width, 1, inputs.get_shape().as_list()[-1]])
    next_inputs = tf.concat([cur_inputs, next_inputs], axis=1)
    next_inputs = tf.concat([next_inputs, zeros_padding], axis=1)
    next_inputs.set_shape([self.config.beam_width, self.params['max_decode_length'], inputs.get_shape().as_list()[-1]])
    outputs = ConvDecoderOutput(
        logits=logits,
        predicted_ids=sample_ids)
    return outputs, enc_output, next_inputs, finished`

Can somebody please have a look at the encoder that I have made in caffe...based on my understanding of the code.

[The caffe_conv_encoder_visualisation](http://ethereon.github.io/netscope/#/gist/cc5ff9df4cdcdbbd2516b320dbed582f)

Ignore the AlexNet part that is there as I am trying to find a sequence in an image 

The network after pool5 is my encoder.  

Any help would be appreciated :) 

Cheers 
Harsh Agarwal
